File: Assignment_13/question2.c
////////////////////////////////////////////////////////////////////////////////
//
//  File Name   : question2.c
//  Description : Display pattern of capital and small letters alternately.
//
//  Problem     : Accept number of rows and columns and display pattern.
//
//  Author      : Asawari Jadhav
//  Date        : 20/05/2025
//
////////////////////////////////////////////////////////////////////////////////

#include <stdio.h>

void Pattern(int iRow, int iCol)
{
  int i = 0;
  int j = 0;
  char ch = '\0';

  for (i = 0; i < iRow; i++)
  {
    if (i % 2 == 0)
    {
      ch = 'A';
    }
    else
    {
      ch = 'a';
    }

    for (j = 0; j < iCol; j++)
    {
      printf("%c\t", ch);
      ch++;
    }
    printf("\n");
  }
}
int main()
{
  int iValue1 = 0, iValue2 = 0;

  printf("Enter number of rows and columns\n");
  scanf("%d %d", &iValue1, &iValue2);

  Pattern(iValue1, iValue2);

  return 0;
}

File: Assignment_50/question2.java
//////////////////////////////////////////////////////////////////////////////////////////////
//
//  File Name     : question2.java
//  Description   : Accept number of rows and columns from user and display pattern of *, # and @
//  Problem       : Display * in initial positions, # and @ on diagonal positions
//
//  Author        : Asawari Jadhav
//
//  Date          : 21-08-2025           
//
//////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Pattern 
{
    public void DisplayPattern(int iRow, int iCol) 
    {
        // Loop through each row
        for (int i = 0; i < iRow; i++) 
        {
            // Loop through each column
            for (int j = 0; j < iCol; j++) 
            {
                if (i == 0 && j == iCol - i - 1) 
                {
                    System.out.print("# ");
                } 
                else if (j == iCol - i - 1) 
                {
                    System.out.print("@ ");
                } 
                else 
                {
                    System.out.print("* ");
                }
            }
            System.out.println();
        }
    }
}

public class question2 
{
    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);

        // Input number of rows and columns
        System.out.print("Enter number of rows: ");
        int iRow = sc.nextInt();
        System.out.print("Enter number of columns: ");
        int iCol = sc.nextInt();


        Pattern pobj = new Pattern();
        pobj.DisplayPattern(iRow, iCol);

    } 
}
